### ISSUE TITLE
Issue with processONT step

### ISSUE THREAD

**Issue by projectoriented** [marbl/verkko#101]
Hello,

Thank you for this tool. I'm experiencing issues with the processONT step and I cannot find what the problem is. Any help would be appreciated:

```
[Thu Sep  1 09:01:07 2022]
rule processONT:
    input: 2-processGraph/unitig-unrolled-hifi-resolved.gfa, 1-buildGraph/hifi_nodecov.csv, 3-align/alns-ont.gaf
    output: 4-processONT/unitig-unrolled-ont-resolved.gfa, 4-processONT/gaps-ont.gaf, 4-processONT/nodecovs-ont.csv
    log: 4-processONT/processONT.err
    jobid: 2
    reason: Missing output files: 4-processONT/gaps-ont.gaf, 4-processONT/nodecovs-ont.csv, 4-processONT/unitig-unrolled-ont-resolved.gfa
    resources: mem_mb=9198, disk_mb=9198, tmpdir=/tmp, job_id=1, n_cpus=1, mem_gb=80, time_h=36

Submitted job 2 with external jobid '289012398'.
Waiting at most 60 seconds for missing files.
MissingOutputException in line 30 of verkko/1.1/Linux/CentOS7/x86_64/envs/verkko/lib/verkko/Snakefiles/4-processONT.sm:
Job Missing files after 60 seconds. This might be due to filesystem latency. If that is the case, consider to increase the wait time with --latency-wait:
4-processONT/unitig-unrolled-ont-resolved.gfa
4-processONT/gaps-ont.gaf
4-processONT/nodecovs-ont.csv completed successfully, but some output files are missing. 2
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2022-09-01T085942.702543.snakemake.log
```

**Comment by brianwalenz** [comment]
Look for messages in 4-processONT/processONT.err and also in the last .err file created in that directory.  Paste those messages here, and also the tail end of `ls -ltr` in the same directory (so we can figure out how far it progressed in that step).

**Comment by skoren** [comment]
My guess is same as #99 but please check for the specific error as @brianwalenz suggested so we can confirm.

**Comment by projectoriented** [comment]
Hello 👋 ! Apologies for the late reply. `4-processONT/processONT.err` is empty.

These are the files in the directory:
```
total 20G
drwxrwsr-x 6 8.0K Aug 23 22:21 0-correction
-rw-rw-r-- 1  20G Aug 23 22:21 hifi-corrected.fasta.gz
drwxrwsr-x 2 4.0K Aug 24 03:50 1-buildGraph
drwxrwsr-x 2 4.0K Aug 26 10:08 2-processGraph
drwxrwsr-x 3  16K Aug 27 18:22 3-align
drwxrwsr-x 2 4.0K Sep  1 09:10 log
-rw-rw-r-- 1 5.7K Sep  1 09:11 verkko.yml
-rwxrwxr-x 1  706 Sep  1 09:11 snakemake.sh
drwxrwsr-x 2  32K Sep  1 09:11 batch-scripts
drwxrwsr-x 2 4.0K Sep  1 09:22 4-processONT
```

And in the 4-processONT
```
total 0
-rw-rw-r-- 1 0 Sep  1 09:12 processONT.err
```

**Comment by skoren** [comment]
Sounds like snakemake being too smart for its own good. Try running the processONT.sh step by hand on a compute node from inside the 4-processONT folder (`sh processONT.sh`) and report the output it produces.

**Comment by projectoriented** [comment]
I found the bug 👾 !
```
Building DAG of jobs...
WorkflowError in line 24 of path/to/lib/verkko/Snakefiles/2-processGraph.sm:
Resource n_cpus is of type str but global resource constraint defines an int with value 1. Resources with the same name need to have the same types (int, float, or str are allowed).
```
There is discordance between the type declared in verkko.yml vs. what's declared in the actual snakefile. What's a good way to fix this?

(as for `sh processONT.sh`, it successfully executed when manually done and I have all the necessary output files)

**Comment by brianwalenz** [comment]
See issue #95 for fixes and patches.

Where was that message reported?  In one of the `batch-scripts/` log files, or after you manually ran processONT.sh and resumed verkko?

**Comment by projectoriented** [comment]
The error message was reported in a cluster node's stdout + stderr log file. This was not an issue in verkko v1.0 and I think it has something to do with `--sge` option. I ended up running the job locally and it ran through. I'll circle back when I have a better solution for this- whether it's caused by our workload manager or verkko itself.

**Comment by skoren** [comment]
The resources aren't going to be queried in the same way when you're not on SGE which would explain why it works without the flag. Either way, the fixes in #95 should address your run as well.
